samples/shared/SynthSound.h

```diff
@@ -25,22 +25,21 @@ constexpr int32_t kDefaultSampleRate = 48000;
 constexpr double kPi = M_PI;
 constexpr double kTwoPi = kPi * 2;
 constexpr int32_t kNumSineWaves = 5;
-constexpr int32_t kNumExtraReleasePeriods = 5;
-constexpr int32_t kAttackPeriods = 40;
+constexpr double kSustainMultiplier = 0.99999;
+constexpr double kReleaseMultiplier = 0.999;
+// Stop playing music below this cutoff
+constexpr double kMasterAmplitudeCutOff = 0.01;
 
 class SynthSound : public IRenderableAudio {
 
 public:
     SynthSound() {
-        for (int i = 0; i < kNumSineWaves; i++) {
-            mFrequencies[i] = kDefaultFrequency * (i + 1);
-        }
     }
     ~SynthSound() = default;
 
     void setWaveOn(bool isWaveOn) {
         if (!mIsWaveOn && isWaveOn) {
```

Comment:
This edge detection should occur at the UI level. Then you could use the same UI to trigger MIDI events, which do not use the set/get paradigm.

If a noteOn() occurs while a note is already on then it should retrigger the envelope. This may not be obvious if using a steady sustain value.

Reply:
Thanks for giving me a suggested fix below. I have done the following instead:

    void noteOn() {
        mTrigger = true; // start a note envelope
        mAmplitudeScaler = kSustainMultiplier;
    }

    void noteOff() {
        mAmplitudeScaler = kReleaseMultiplier;
    }